### PR TITLE
regression tests for evince and gedit

### PR DESCRIPTION
**Pull request by mitiao**

### CODE REVIEW

File: tests/x11regressions/evince/evince_find.pm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    for ( 1..2 ) {
+        send_key "ctrl-shift-g";    # go to previous result
+        sleep 1;
+    }
+    assert_screen 'evince-search-1stresult', 5;
+    
+    send_key "esc";
+    sleep 1;
+    send_key "ctrl-w";
+    sleep 1;
```
Comment (nilxam) [inline tests/x11regressions/evince/evince_find.pm]: this is unnecessary, x11test will check assert_screen('generic-desktop') in the end ie. the post hook.

**Comment by nilxam** [comment]
1) those +sleep 1+ at the end of run() looks unnecessary to me.
2) many +sleep+ just following +send_key+ looks unnecessary because it has assert_screen() in the next step, the real case need to wait until system idle might consider to use +sendkey $key, 1+ which will call wait_idle inside of send_key

just my 2 cents,

File: tests/x11regressions/evince/evince_find.pm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Case 1436022 - Evince: Find
+sub run() {
+    my $self = shift;
+    x11_start_program("evince " . autoinst_url . "/data/x11regressions/test.pdf");
+
+    send_key "ctrl-f";    # show search toolbar
+    assert_screen 'evince-search-toolbar', 5;
+
+    my $keyword = 'To search for';
+    type_string $keyword;
```
Comment (aaannz) [inline tests/x11regressions/evince/evince_find.pm]: Why are you using $keyword variable when it's only used once? Why not
```perl
type_string 'To search for';
```

Am I missing something?

**Comment by aaannz** [comment]
I agree with @nilxam here. These sleeps seem unnecessary to me. If you need to wait after send_key, you can pass `1` as second, optional, parameter.
```perl
send_key 'esc', 1
```
this will trigger `wait_idle` after key is pressed.

File: tests/x11regressions/evince/evince_find.pm (hunk 2 above)
Comment (mitiao) [inline tests/x11regressions/evince/evince_find.pm]: you are right, no need to set variable here

**Comment by mitiao** [comment]
will remove unnecessary sleep :)

**Comment by sysrich** [comment]
Looks good, can you please rebase using something like "git rebase -i HEAD~4" to 'squash' the extra commit messages?

After squashing the 3 extra commit messages you will then need to "git push --force", but it should retain all the changes

we don't need to see all 4 commits :)

**Comment by mitiao** [comment]
Done

**Comment by mitiao** [comment]
i think this was already merged, but today i found the files were not in master, any problem?

**Comment by nilxam** [comment]
hmm...weird

**Comment by nilxam** [comment]
I've pick this patch and re-merge to master, hopefully fixed the problem, please check again.

**Comment by aaannz** [comment]
Do we know what happened? Someone force pushed to hide some embarrassing stuff from history? :)

**Pull request merged**